### ISSUE TITLE
What fasta do folks use as blast input? 

### ISSUE THREAD

**Issue by laurahspencer** [RobertsLab/resources#1898]
Sam/Steven asked me exactly which .fasta file I use as input when blasting. I've used a couple different approaches when needing to get Uniprot annotation information for genomes, so curious what others use, and if you all have opinions on the best approach? Or are there other approaches?  

  Option 1. Use `bedtools getFasta` + .bed file of gene locations (chromosome, start, stop) + genome fasta to pull gene sequences from the genome. The blast input is a fasta with one full sequence for each gene based on the genome. 
  
  Option 2. Use a .cds fasta file for the genome (if available). For example, for the [Pacific cod genome](https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_031168955.1/) I've downloaded the "Genomic coding sequences" fasta file (screen shot below) and used that as input. The .cds fasta contains multiple sequences for some genes (when there are multiple isoforms). There are multiple blast hits for the same gene, so when I'm analyzing data at the gene level I keep the blast hit with the smallest e-value. 

Option 2 definitely results in more Uniprot-annotated genes passing my e-value filter (1e-20). 

<img src="https://github.com/RobertsLab/resources/assets/17264765/29dcbe69-02d5-461e-a2db-757844625ede" width="450">

**Comment by sr320** [comment]
I am going to read a lot into this and presume this is related to enrichment. 

Why not start from your count matrix after filtering, and determine what those IDs are, then back track to find the appropriate manner to get corresponding fastas. This way you have correct background and correct fasta in one step. 

I did a similar approach here: https://robertslab.github.io/tusk/modules/04-blast.html#case-study---oyster 

reminiscent of https://github.com/RobertsLab/resources/issues/1897

In short, it would depend on your approach and setting to get your count matrix (what you feed into your differential gene expression package.

**Comment by kubu4** [comment]
> Sam/Steven asked me exactly which .fasta file

I think we were specifically asking about the cod FastA (at least, I was) for the cod project, to compare with what Kathleen had used.

**Comment by laurahspencer** [comment]
I'm thinking this is a general question about blast- when needing to annotate genes with Uniprot IDs (for enrichment, or just to be able to describe gene functions and learn more about them via the Uniprot resources) is it better to blast entire gene sequences that are pulled from a genome, OR blast coding sequences only? 

Given that many of Uniprot's protein sequences are translated from coding sequences (according to [this page](https://www.uniprot.org/help/sequence_origin), my guess is that it's best to use coding sequences! Perhaps using the full gene sequence during blast would decrease hit rate due to presence of introns (?). Given that many of Uniprot's protein sequences are translated from coding sequences (according to [this page](https://www.uniprot.org/help/sequence_origin), my guess is that it's best to use coding sequences! Perhaps using the full gene sequence during blast would decrease hit rate due to presence of introns (?).

**Comment by sr320** [comment]
Use the protein fasta!
